Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -14,16 +14,99 @@ static void Main(string[] args)
             Console.ReadKey(true);
         }
 
-        private static void Test(string args, int expected)
-        {
+        private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+            Console.WriteLine($"{args} : {result}");
         }
-
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        //function to check missing integer
+        public static bool IsMissing(string num){
+            for(int i=0; i<num.Length; i++){
+                //checking for quetion mark
+                if(num[i]=='?')
+                    return true;
+            }
+            return false;
+        }
+        //function to find ans in case missing integer is a multiplier...
+        public static int FindMissingInteger(string missingNumber, string helpingNumber, string realRes, string result, bool flag){
+            if(missingNumber.Length == result.Length){
+                int len = missingNumber.Length;
+                int count = 0, ans = -1;
+                //comparing result and the missing number 
+                string k = "";
+                for(int i = 0; i < len; i++){
+                    //counting total unequal digits both numbers 
+                    if(missingNumber[i] != result[i]) count++;
+                    //checking for question-mark
+                    if(missingNumber[i] == '?'){
+                        ans = result[i]-'0';
+                    }           
+                        //generating new value of k after eleminating '?' for calculations            
+                    k+=result[i];
+                }
+                bool b;
+                if(flag){
+                    //comparing result in case of multiplier digit is missing.
+                    b = (int.Parse(k)*int.Parse(helpingNumber) == int.Parse(realRes));
+                } else
+                {
+                    //comparing result in case of results digit is missing.
+                     b = int.Parse(helpingNumber) * int.Parse(realRes) == int.Parse(k);
+                }
+                //returning result if only one digit is unequal and b is true.
+                // return -1
+                return count == 1 && b ? ans : -1;
+            }
+            //else return -1.
+            return -1;
+        }
+        public static int FindDigit(string equation)

```

Comment:
Your Business Logic should be in a separate class.
And call that method from Program class

Reply:
New changes has been commited kindly review again